### PR TITLE
groovy-language-server: init at 2023-03-20

### PR DESCRIPTION
**Pull request by traxys**
## Description of changes

Add a language server for groovy, [groovy-language-server](https://github.com/GroovyLanguageServer/groovy-language-server/)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by traxys** [comment]
Result of `nixpkgs-review pr 250945` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>groovy-language-server</li>
  </ul>
</details>

File: pkgs/development/tools/groovy-language-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    gradle --offline --no-daemon build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ls -la build/libs
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/groovy-language-server/default.nix:67]:
```suggestion
```

File: pkgs/development/tools/groovy-language-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gradle_7
+, perl
+, makeWrapper
+, openjdk17_headless
+,
```
Comment (SuperSandro2000) [inline pkgs/development/tools/groovy-language-server/default.nix:8]:
```suggestion
```

File: pkgs/development/tools/groovy-language-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}:
+let
+  pname = "groovy-language-server";
+  version = "2023-03-20";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/groovy-language-server/default.nix:12]:
```suggestion
  version = "unstable-2023-03-20";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
somehow ofborg fails. Can you please take a look at this?

**Comment by traxys** [comment]
Sorry I'm not sure where to see that failure ?

**Comment by SuperSandro2000** [comment]
image

which then leads to

https://logs.ofborg.org/?key=nixos/nixpkgs.250945&attempt_id=4392a35e-19be-4055-8f09-6c1f83b94fb7

```
<-------------> 0% EXECUTING [7s]> IDLE> IDLE> :compileJava> IDLE<-------------> 0% EXECUTING [8s]<=====--------> 41% EXECUTING [8s]> :shadowJar<=====--------> 41% EXECUTING [9s]<=====--------> 41% EXECUTING [10s]<======-------> 50% EXECUTING [10s]> :compileTestJava > Resolve dependencies of :testCompileClasspath > junit-bom-> :compileTestJava
> Task :compileTestJava
Note: /build/source/src/test/java/net/prominic/groovyls/util/FileContentsTrackerTests.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
<========-----> 66% EXECUTING [10s]> IDLE> IDLE> IDLE> IDLE<=========----> 75% EXECUTING [10s]> :test > 0 tests completed<=========----> 75% EXECUTING [11s]> :test > Executing test net.prominic.groovyls.GroovyServicesCompletionTests<=========----> 75% EXECUTING [12s]> :test > 1 test completed> :test > 2 tests completed> :test > 3 tests completed<=========----> 75% EXECUTING [13s]> :test > 4 tests completed> :test > 5 tests completed> :test > 6 tests completed> :test > 7 tests completed> :test > 8 tests completed> :test > 9 tests completed<=========----> 75% EXECUTING [14s]> :test > 10 tests completed> :test > 11 tests completed> :test > 12 tests completed> :test > 13 tests completed> :test > 14 tests completed> :test > 15 tests completed> :test > 16 tests completed<=========----> 75% EXECUTING [15s]> :test > 17 tests completed> :test > 18 tests completed> :test > 19 tests completed> :test > Executing test net.prominic.groovyls.GroovyServicesDefinitionTests> :test > 20 tests completed> :test > 21 tests completed> :test > 22 tests completed> :test > 23 tests completed<=========----> 75% EXECUTING [16s]> :test > 24 tests completed> :test > 25 tests completed> :test > 26 tests completed> :test > 27 tests completed> :test > 28 tests completed> :test > 29 tests completed> :test > 30 tests completed<=========----> 75% EXECUTING [17s]> :test > 31 tests completed> :test > 32 tests completed> :test > 33 tests completed> :test > 34 tests completed> :test > 35 tests completed> :test > 36 tests completed> :test > Executing test net.prominic.groovyls.GroovyServicesSignatureHelpTests> :test > 37 tests completed<=========----> 75% EXECUTING [18s]> :test > 38 tests completed> :test > 39 tests completed> :test > Executing test net.prominic.groovyls.GroovyServicesTypeDefinitionTest> :test > 40 tests completed> :test > 41 tests completed> :test > 42 tests completed> :test > 43 tests completed> :test > 44 tests completed<=========----> 75% EXECUTING [19s]> :test > 45 tests completed> :test > 46 tests completed> IDLE> :test
BUILD SUCCESSFUL in 20s
5 actionable tasks: 5 executed
<-------------> 0% WAITING> IDLE> IDLE> :test> IDLE@nix { "action": "setPhase", "phase": "installPhase" }
installing
sh: line 112: /build/tmp.A5hrYQwJsN/caches/7.6.2/workerMain/gradle-worker.jar: Permission denied
error: builder for '/nix/store/8hlgi44d450nxxvhnlvjaiqii7crwhyb-groovy-language-server-deps-unstable-2023-03-20.drv' failed with exit code 126;
```

**Comment by traxys** [comment]
I switched to using `gradle_6` as it seems that's what upstream is using. It builds fine on my machine, let's see if ofborg is happy with gradle_6

**Comment by traxys** [comment]
I don't understand at all what is happening, the derivation builds nicely on my machine, but does not with ofborg. In addition it seems that the hashes are not the same between aarch64 & x86_64 and I'm not really sure how to fix that.

I'm more inclined to drop this, as packaging java seems a big problem...

**Pull request closed without merging**